viewer-prototype/src/browser/trace-viewer/components/timegraph-output-component.tsx

```diff
@@ -43,13 +46,25 @@ export class TimegraphOutputComponent extends AbstractTreeOutputComponent<Timegr
         super(props);
         this.state = {
             outputStatus: ResponseStatus.RUNNING,
-            timegraphTree: []
+            timegraphTree: [],
+            collapsedNodes: []
         };
         this.tspDataProvider = new TspDataProvider(this.props.tspClient, this.props.traceId, this.props.outputDescriptor.id);
         this.rowController = new TimeGraphRowController(this.props.style.rowHeight, this.totalHeight);
         this.horizontalContainer = React.createRef();
+
+        this.vscrollLayer = new TimeGraphVerticalScrollbar('timeGraphVerticalScrollbar', this.rowController);
+
+        this.rowController.onVerticalOffsetChangedHandler(()=>{
+            if (this.treeRef.current) {
+                this.treeRef.current.scrollTop=this.rowController.verticalOffset;
+            }
+        });
+    }
+
+    createChartLayer(orderedTreeIds: number[]): void {
         const providers: TimeGraphChartProviders = {
-            dataProvider: async (range: TimelineChart.TimeGraphRange, resolution: number) => this.fetchTimegraphData(range, resolution),
+            dataProvider: async (viewRange: TimelineChart.TimeGraphRange, resolution: number) => this.fetchTimegraphData(orderedTreeIds, viewRange, resolution),
```

Comment:
Talking to Soukaina here:

here should be

async (orderedTreeIds: number[], viewRange: TimelineChart.TimeGraphRange, resolution: number)

and the chart layer should be created only once. It's the data provider function call that needs to add a parameter I think.

So in the time-graph-chart.d.ts of the timeline-chart library, the TimeGraphChartProviders should be a function that also adds an array of entry ids.